src/vs/workbench/contrib/terminal/browser/links/terminalValidatedLocalLinkProvider.ts

```diff
@@ -59,13 +61,19 @@ export class TerminalValidatedLocalLinkProvider extends TerminalBaseLinkProvider
 		private readonly _wrapLinkHandler: (handler: (event: MouseEvent | undefined, link: string) => void) => XtermLinkMatcherHandler,
 		private readonly _tooltipCallback: (link: TerminalLink, viewportRange: IViewportRange, modifierDownCallback?: () => void, modifierUpCallback?: () => void) => void,
 		private readonly _validationCallback: (link: string[], callback: (result: { uri: URI, link: string, isDirectory: boolean } | undefined) => void) => void,
+		private readonly _testing: boolean,
```

Comment:
We could instead make this `protected static _enableCaching = true` and then create a `TestTerminalValidatedLocalLinkProvider` which overrides it, less ctor args is always better